js/common/modules/@arangodb/graph/pregel-test-helpers.js

```diff
@@ -436,6 +535,293 @@ class PageRank {
+            let maxDiffAuth = 0.0;
+            for (let [, vertex] of graph.vertices) {
+                let sumAuth = 0.0;
+                for (const predKey of vertex.inNeighbors) {
+                    const pred = graph.vertex(predKey);
+                    assertTrue(pred.hasOwnProperty("value"));
+                    sumAuth += pred.value.old_hits_hub;
+                }
+                vertex.value.hits_auth = sumAuth / authDivisor;
+                maxDiffAuth = Math.max(maxDiffAuth, Math.abs(vertex.value.hits_auth - vertex.value.old_hits_auth));
+            }
+
+            return {maxDiffAuth};
+        };
+
+        const computeHub = function (hubDivisor) {
+            let maxDiffHub = 0.0;
+            for (let [, vertex] of graph.vertices) {
+                let sumHub = 0.0;
+                for (const succKey of vertex.outNeighbors) {
+                    const succ = graph.vertex(succKey);
+                    assertTrue(succ.hasOwnProperty("value"));
+                    sumHub += succ.value.old_hits_auth;
+                }
+
+                vertex.value.hits_hub = sumHub / hubDivisor;
+                maxDiffHub = Math.max(maxDiffHub, Math.abs(vertex.value.hits_hub - vertex.value.old_hits_hub));
+            }
+
+            return {maxDiffHub};
+        };
+
+        const getAuthDivisorFromHub = function () {
+            let divisor = 0.0;
+            for (const [, v] of graph.vertices) {
+                divisor += v.value.hits_hub * v.value.hits_hub;
+            }
+            return Math.sqrt(divisor);
+        };
+
+        const getHubDivisorFromAuth = function () {
+            let divisor = 0.0;
+            for (const [, v] of graph.vertices) {
+                divisor += v.value.hits_auth * v.value.hits_auth;
+            }
+            return Math.sqrt(divisor);
+        };
+
+        const saveOldValues = function () {
+            for (let [, v] of graph.vertices) {
+                v.value.old_hits_auth = v.value.hits_auth;
+                v.value.old_hits_hub = v.value.hits_hub;
+            }
+        };
+
+        assertTrue(epsilon > 0);
+        assertTrue(numSteps > 0);
+
+        let step = 0;
+        let maxDiff = 0.0;
+        do {
+            maxDiff = 0.0;
+            const authDivisor = getAuthDivisorFromHub();
+            const hubDivisor = getHubDivisorFromAuth();
+            saveOldValues();
+            const {maxDiffAuth} = computeAuth(authDivisor);
+            const {maxDiffHub} = computeHub(hubDivisor);
+            maxDiff = Math.max(maxDiff, maxDiffAuth, maxDiffHub);
+            ++step;
+        } while (step < numSteps && maxDiff > epsilon);
+
+        return maxDiff;
+    }
+
+    /**
+     * Perform one step of the HITS algorithm and return the new value of the given vertex.
+     * @returns {*}
+     * @param vKey vertex key
+     * @param graph a graph containing a vertex with this key
+     */
+    doOneHITSStep(vKey, graph) {
```

Comment:
which HITS algorithm is that?

Reply:
It was our older version, but we don't test it anymore. I will delete the function.